### DISCUSSION TITLE
Hello, respected author! I would like to ask you a question about DTN7 networking

### DISCUSSION THREAD

**Discussion by blueberrylxm** [dtn7/dtn7-rs#62] [Q&A]
Hello, respected author! I would like to ask you a question about DTN7 networking
I am conducting a networking experiment using DTN7 for virtual machines and Raspberry Pi. I can discover neighbors with each other, but messages cannot be transmitted to each other. I hope to receive the help of a respected author.

For example：

"nY3NKYLao7":{
"eid": [1,
"//nY3NKYLao7/"]，
"addr": {
"Ip": "192.168.67.242"},
"con_type": "Dynamic","period": null,"cla_list": []，"services": {},
"last contact": 1709969019，"fails": 0

Lu@lu-virtual-machine:~/.cargo/bin$./dtnquery eids isting registered endpoint IDs:
"dtn://BlsyxdKnPd/"

My computer can only transmit messages to this device, and messages transmitted to other devices cannot reach it.

Best wishes!

**Comment by gh0st42** [comment]
Hi,

it seems like you did not configure a convergence layer to connect the two nodes.
How did you start the two dtnd's?

You might wanna try this for startup: `dtnd -n $(hostname) -C mtcp -r epidemic`

The important part here is `-C` followed by the CL, something like `mtcp`, `tcp` or `http` can be used and all have different pros and cons.

I hope this helps.

PS: More usage examples can be found in the [getting started guide](https://github.com/dtn7/dtn7-rs/blob/master/doc/getting-started.md)... but it might be a bit outdated in some parts and does not show the more advanced commands.

PPS: The integration tests under `tests/` also can serve as templates on how to configure different stuff.
